src/lib.rs

```diff
@@ -0,0 +1,540 @@
+
+        // pub value_length: usize, //8 bytes
+        // pub value: *mut u8, //value_length bytes
+    }
+
+    // pub type BlockHeightLookup = unsafe extern "C" fn(block_hash: [u8; 32]) -> u32;
+    // pub type MasternodeListLookup = unsafe extern "C" fn(block_hash: [u8; 32]) -> Option<MasternodeList>;
+    // pub type AddInsightBlockingLookup = unsafe extern "C" fn(block_hash: [u8; 32]) -> bool;
+
+    pub type BlockHeightLookup = unsafe extern "C" fn(block_hash: *const u8) -> u32;
+    pub type MasternodeListLookup = unsafe extern "C" fn(block_hash: *const u8) -> BaseMasternodeList;
+    pub type AddInsightBlockingLookup = unsafe extern "C" fn(block_hash: *const u8) -> bool;
+    pub type ShouldProceeQuorumTypeCallback = unsafe extern "C" fn(quorum_type: LLMQType) -> bool;
+    pub type ValidateOperatorCallback = unsafe extern "C" fn(public_key_signatures: Vec<UInt384>) -> bool;
+    pub type ValidateOperatorPublicKeySignaturesFunc = unsafe extern "C" fn(
+        items: *mut *mut UInt384, // 8 bytes
+        count: usize // sizeof(pointer)
+    ) -> bool;
+
+
+    //pub struct BaseMasternodeList(pub Option<MasternodeList>);
+
+    #[repr(C)]
+    #[derive(Debug)]
+    pub struct BaseMasternodeList(pub Option<*mut MasternodeList<'static>>);
+
+    /*pub fn block_until_add_insight(block_hash: &[u8; 32], chain: Chain) {
+        assert_ne!(block_hash, 0);
+        let insight_url = if chain.is_main_net() {INSIGHT_URL} else {TESTNET_INSIGHT_URL};
+        let path = format!("{}{}{}", insight_url, BLOCK_PATH, hex_with_data(block_hash)).as_str();
+        let client = Client::new();
+        let uri = path.parse()?;
+        let mut resp = client.get(uri).await?;
+        println!("Response: {}", resp.status());
+        let mut body = Vec::new();
+        while let Some(chunk) = resp.body_mut().next().await {
+            body.extend_from_slice(&chunk?);
+        }
+        let block_data: BlockData = serde_json::from_slice(&body)?;
+        chain.add_insight_verified_block(block_data);
+    }*/
+
+    fn boxed<T>(obj: T) -> *mut T { Box::into_raw(Box::new(obj)) }
+
+    fn failure<'a>() -> *mut Result<'a> {
+        boxed(Result {
+            found_coinbase: false,
+            valid_coinbase: false,
+            root_mn_list_valid: false,
+            root_quorum_list_valid: false,
+            valid_quorums: false,
+            masternode_list: BaseMasternodeList(None),
+            added_masternodes: None,
+            modified_masternodes: None,
+            added_quorums: None,
+            needed_masternode_lists: None,
+        })
+    }
+
+    #[no_mangle]
+    pub extern fn process_diff(
+        c_array: *const u8,
+        length: usize,
+        base_masternode_list: BaseMasternodeList,
+        masternode_list_lookup: MasternodeListLookup,
+        merkle_root: *const u8,
+        use_insight_lookup: AddInsightBlockingLookup,
+        should_process_quorum_of_type: ShouldProceeQuorumTypeCallback,
```

Comment:
`ShouldProceeQuorumTypeCallback` should be `ShouldProceedQuorumTypeCallback`?

Reply:
+